worker/src/main/java/gov/cms/ab2d/worker/processor/JobProcessorImpl.java

```diff
@@ -111,65 +149,290 @@ public Job process(final String jobUuid) {
     /**
      * Process in individual contract
      *
-     * @param contract - the contract to process
+     * @param contracts - all the contract to process
      * @param job - the job in which the contract belongs
      * @param month - the month to search for beneficiaries for
      * @param outputDirPath - the location of the job output
      * @param progressTracker - the progress tracker which indicates how far the job is along
      * @throws ExecutionException when there is an issue with searching
      * @throws InterruptedException - when the search is interrupted
      */
-    void processContract(Contract contract, Job job, int month, Path outputDirPath, ProgressTracker progressTracker) throws ExecutionException, InterruptedException {
-        log.info("Job [{}] - contract [{}] ", job.getJobUuid(), contract.getContractNumber());
-        // Retrieve the contract beneficiaries
+    private void processContracts(List<Contract> contracts, Job job, int month, Path outputDirPath,
+                                  ProgressTracker progressTracker) throws Exception {
+        List<ContractData> cData = new ArrayList<>();
+        List<JobOutput> jobOutputs = new ArrayList<>();
+        List<Integer> monthsDone = new ArrayList<>();
+
+        for (Contract contract : contracts) {
+            String contractNum = contract.getContractNumber();
+            progressTracker.addContract(contractNum);
+            ContractEobManager contractEobManager = new ContractEobManager(fhirContext, skipBillablePeriodCheck,
+                    getStartDate(contractNum), contract.getAttestedOn());
+
+            // Retrieve the contract beneficiaries
+            //for (Contract contract : contracts) {
+            try {
+                // Init objects
+                StreamHelper helper = new TextStreamHelperImpl(outputDirPath, contractNum, ndjsonRollOver * Constants.ONE_MEGA_BYTE, tryLockTimeout, eventLogger, job);
+                ContractData contractData = new ContractData(contract, progressTracker, contract.getAttestedOn(), job.getSince(),
+                        job.getUser() != null ? job.getUser().getUsername() : null, helper);
+                cData.add(contractData);
+
+                ContractBeneficiaries contractBeneficiaries = buildCB(contractNum);
+
+                progressTracker.setCurrentMonth(month);
+
+                List<Future<ContractMapping>> contractBeneFutureHandles = createAllContractMappingFutures(month, contractNum, contract.getAttestedOn());
+                List<Future<EobSearchResponse>> eobFutureHandles = new ArrayList<>();
+
+                 // Iterate through the futures
+                while (!contractBeneFutureHandles.isEmpty() || !eobFutureHandles.isEmpty()) {
```

Comment:
Could contractBeneFutureHandles be empty before the eobFutreHandles is, leaving eobFutureHandles that are unprocessed?

Reply:
No, but maybe:
! list1.isEmpty() || ! list2.isEmpty()
is not as clear as:
list1.size() > 0 || list2.size() > 0 although I think isEmpty() is in theory more efficient

Reply:
isEmpty() is absolutely more efficient for certain kinds of collections (such as a queue whose size is constantly fluctuation) and is strongly preferred for existence checks.

Reply:
It vastly depends upon the implementation of the Collection.

Reply:
Technically, it's true, for ArrayList, it is the same because the implementation maintains the size as an independent variable and updates it through removes and adds, but in terms of best practices, it is recommended to use isEmpty. I'm guessing it's because you may end up changing the underlying collection for some reason. Everytime I use size(), some plugin of IntelliJ complains.